pandas/_libs/algos.pyx

```diff
@@ -931,6 +931,32 @@ ctypedef fused rank_t:
     int64_t
 
 
+cdef rank_t get_rank_nan_fill_val(bint rank_nans_highest, ndarray[rank_t, ndim=1] _):
```

Comment:
Does the second argument here need to be an ndarray or can it be a `type`?

Reply:
Thanks for the great suggestion! `type` makes sense as a way to allow the `rank_t` parameterization - messed around with this for a while trying to use `type[rank_t]`, but kept running into failures when actually trying to define this value. For example, I wanted to do something like  `cdef type[rank_t] = values.dtype.type`, but kept running into unhelpful errors (maybe this definition just doesn't make sense, but it compiled).

But just pushed a much cleaner solution than the previous way of handling this type specialization.